Sources/ComposableArchitecture/Reducer/Reducers/CombineReducers.swift

```diff
@@ -0,0 +1,50 @@
+/// Combines multiple reducers into a single reducer.
+///
+/// `CombineReducers` takes a block that can combine a number of reducers using a
+/// ``ReducerBuilder``.
+///
+/// Useful for grouping reducers together and applying reducer modifiers to the result.
+///
+/// ```swift
+/// var body: some ReducerProtocol<State, Action> {
+///   CombineReducers {
+///     ReducerA()
+///     ReducerB()
+///     ReducerC()
+///   }
+///   .ifLet(\.child, action: /Action.child)
+/// }
+/// ```
+public struct CombineReducers<Reducers: ReducerProtocol>: ReducerProtocol {
+  @usableFromInline
+  let reducers: Reducers
+
+  /// Initializes a reducer that combines all of the reducers in the given build block.
+  ///
+  /// - Parameter build: A reducer builder.
+  @inlinable
+  public init(
+    @ReducerBuilderOf<Reducers> _ build: () -> Reducers
+  ) {
+    self.reducers = build()
```

Comment:
I am getting the following compiler error on this line:
CombineReducers.swift:29:10: error: 'let' property 'reducers' may not be initialized directly; use "self.init(...)" or "self = ..." instead

Reply:
Can you share which Xcode and Swift version you are using?

Reply:
This error only happens if library evolution is enabled (BUILD_LIBRARY_FOR_DISTRIBUTION=YES)

Reply:
Basically, we are building ComposableArchitecture into an xcframework (and thus enable the library evolution option). Although I understand that it may not be your intention to support this mode at all, it's still a really nice mode for corporate environments, where direct access to GitHub is disabled.

Reply:
I am getting similar errors for other reducers as well (where "self.reducers = ... " line is present).
Not sure if it's a known compiler limitation or a bug though.

Reply:
The compiler option we are adding is **-enable-library-evolution**

Reply:
Let's start a new discussion for this because I have some questions. Got time?

Reply:
Yes. Definitely!

Reply:
There is also an older discussion (where I contributed a bit previously) here: https://github.com/pointfreeco/swift-composable-architecture/discussions/890

Reply:
As far as I understood, that option had to be enabled to simply produce an xcframework with the proper .swiftinterface file.
As such we don't require binary compatibility between different versions of Composable Architecture. But we still like to be able to compile an xcframework once (on the CI) (for a certain version of Composable Architecture) and then use it for different internal builds.

Reply:
As far as I can tell you will just get a warning if you build an xcframework without library evolution letting you know the resulting binary has no compatibility guarantees. But now this discussion is getting a bit long, so I do think we should move to a separate thread. 🙂

Reply:
Sure. We can continue on https://github.com/pointfreeco/swift-composable-architecture/discussions/890 just create another discussion if you think it's better...

Reply:
I think a separate thread because this is specifically about library evolution. And can you try building without evolution turned on and let me know what happens?

Reply:
https://github.com/pointfreeco/swift-composable-architecture/discussions/1442